src/cmd/ksh93/bltins/print.c

```diff
@@ -375,9 +375,10 @@ int b_print(int argc, char *argv[], Shbltin_t *context) {
         // with syncing history if -s and -f are used together. History is synced
         // later with histflush() function.
         // https://github.com/att/ast/issues/425
-        if (!sflag && sffileno(outfile) != sffileno(sfstderr)) sfsync(outfile);
+        if (!sflag && sffileno(outfile) != sffileno(sfstderr))
+            exitval |= (sfsync(outfile) < 0);
         sfpool(sfstderr, pool, SF_WRITE);
-        exitval = pdata.err;
+        exitval |= pdata.err;
```

Comment:
I dislike this pattern since `exitval` is not a bitmask. Please change all such statements to the equivalent of `if (pdata.err) exitval = 1;`. This is the type of thing that might shave a few microseconds off the run time but cannot be justified. It makes bugs more likely. Not to mention reasoning about the code far harder. If this were a performance critical path I would allow it but that isn't the case.

Reply:
Very well. Done.